ui/src/components/Button.tsx

```diff
@@ -5,11 +5,11 @@ import { JSXElement, Show } from 'solid-js'
  * Renders a button component with an optional loading state.
  * If the loading state is passed the button will be disabled and show
  * a loader ball while loading.
- *
  */
 export function Button(props: {
-  children: JSXElement[] | JSXElement
-  onClick?: () => void
+  label: string
+  icon?: string
+  onClick?: (event?: InputEvent) => void
   isLoading?: boolean
   type?: 'submit' | 'button' | 'reset'
   variant?: 'primary' | 'secondary' | 'error' | 'success'
```

Comment:
I would still change these to match the daisyui docs. So something like:

```
variant?: 'primary' | 'secondary' | 'error' | 'success' | 'neutral' | 'accent' | 'info'
style?: 'outline' | 'dash' | 'soft' | 'ghost' | 'link'
size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl'
```

These types are also reusable for all components so we can define them somewhere else maybe?

Reply:
Ah I see you already have an issue for that #123